addons/account/models/account_move.py

```diff
@@ -1522,6 +1522,8 @@ def _inverse_partner_id(self):
 
     @api.onchange('company_id')
     def _inverse_company_id(self):
+        if not self.company_id:
+            raise UserError(_("You cannot remove a company."))
```

Comment:
Can't we put it as `required=True` on the field declaration? Should work since it is precomputed.
If not, setting it as required in the view should do the trick too. Both a re better for the user experience than this.

Reply:
Hello @william-andre we can do `required=True` but when we remove the company then it's instantly call onchange and we get the traceback, we can avoid with this change here --> https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_move.py#LL3822C42-L3822C68
![image](https://github.com/odoo/odoo/assets/85601944/defde9d1-1174-4cae-83bf-26aa5769c510)

what is your opinion on this?
Thanks.

Reply:
What happens if you return `date.max` in `_get_user_fiscal_lock_date` when `not self`?

Reply:
Hello @william-andre 
I have make changes and it's working properly.

Thanks.